librz/core/canalysis.c

```diff
@@ -4753,26 +4710,24 @@ RZ_API void rz_core_analysis_esil(RzCore *core, const char *str, const char *tar
 				if (!strcmp(op.src[0]->reg->name, "zero")) {
 					break;
 				}
-				if ((target && dst == ntarget) || !target) {
-					if (dst > 0xffff && op.src[1] && (dst & 0xffff) == (op.src[1]->imm & 0xffff) && myvalid(core->io, dst)) {
-						RzFlagItem *f;
-						char *str;
-						if (CHECKREF(dst) || CHECKREF(cur)) {
-							rz_analysis_xrefs_set(core->analysis, cur, dst, RZ_ANALYSIS_REF_TYPE_DATA);
-							if (cfg_analysis_strings) {
-								add_string_ref(core, op.addr, dst);
-							}
-							if ((f = rz_core_flag_get_by_spaces(core->flags, dst))) {
-								rz_meta_set_string(core->analysis, RZ_META_TYPE_COMMENT, cur, f->name);
-							} else if ((str = is_string_at(core, dst, NULL))) {
-								char *str2 = sdb_fmt("esilref: '%s'", str);
-								// HACK avoid format string inside string used later as format
-								// string crashes disasm inside agf under some conditions.
-								// https://github.com/rizinorg/rizin/issues/6937
-								rz_str_replace_char(str2, '%', '&');
-								rz_meta_set_string(core->analysis, RZ_META_TYPE_COMMENT, cur, str2);
-								free(str);
-							}
+				if (dst > 0xffff && op.src[1] && (dst & 0xffff) == (op.src[1]->imm & 0xffff) && myvalid(core->io, dst)) {
```

Comment:
This will need more refactoring than extracting this into a separate function. Especially since we are in core here and RzIL analysis will be in analysis and this is even arch-specific, so should be controlled by the plugin.